sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py

```diff
@@ -698,27 +697,35 @@ def create_or_update(
             if snapshot_id is not None:
                 job_object.properties.properties["ContentSnapshotId"] = snapshot_id
 
-            result = self._operation_2023_02_preview.create_or_update(
-                id=rest_job_resource.name,  # type: ignore
-                resource_group_name=self._operation_scope.resource_group_name,
-                workspace_name=self._workspace_name,
-                body=job_object,
-                **kwargs,
-            )
+            result = self.create_or_update_with_different_version_api(rest_job_resource=job_object, **kwargs)
+
         return self._resolve_azureml_id(Job._from_rest_object(result))
 
+    def create_or_update_with_different_version_api(self, rest_job_resource, **kwargs):  # pylint: disable=name-too-long
```

Comment:
note that this is a public API, end user may access it via `ml_client.jobs.create_or_update_with_different_version_api`. I'm afraid this is not what we want.

Reply:
Yes, it can be modify to private.

Reply:
BTW, where do you test this interface? It seems that you haven't added new e2e test for this

Reply:
and it won't be called in regular `create_or_update` call, right?

Reply:
ignore the 2nd question, double-checked the code and it is called

Reply:
New e2e test is not required because it is only asynchronous on the backend, and there is no difference between it and before in the SDK layer, and the problem cannot be detected.

Reply:
done